libres/lib/analysis/ies/ies.cpp

```diff
@@ -84,37 +84,48 @@ Eigen::MatrixXd ies::makeX(const Eigen::MatrixXd &A, const Eigen::MatrixXd &Y0,
 
     Eigen::MatrixXd Y = Y0;
 
-    double nsc = 1.0 / sqrt(ens_size - 1.0);
-
-    /*  Subtract mean of predictions to generate predicted ensemble anomaly matrix (Line 5) */
-    Y = nsc * (Y.colwise() -
-               Y.rowwise().mean()); // Y = Y * (I - (1 / ens_size) * 11')
+    /* Normalized predicted ensemble anomalies.
+       Line 4 of Algorithm 1, also (Eq. 30) 
+    */
+    Y = (1.0 / sqrt(ens_size - 1.0)) * (Y.colwise() - Y.rowwise().mean());
 
-    /* COMPUTING THE PROJECTION Y= Y * (Ai^+ * Ai) (only used when state_size < ens_size-1) */
+    /* A^+A projection is necessary when the parameter matrix has less rows than columns,
+       and when the forward model is non-linear.
+       Section 2.4.3 
+    */
     if (A.rows() > 0 && A.cols() > 0) {
         const int state_size = A.rows();
         if (use_aa_projection && (state_size <= (ens_size - 1))) {
             ies::linalg_compute_AA_projection(A, Y);
         }
     }
 
-    /*
-     * When solving the system S = Y inv(Omega) we write
-     *   Omega^T S^T = Y^T (line 6)
-     */
-    Eigen::MatrixXd S = ies::linalg_solve_S(W0, Y);
+    /* Line 5 of Algorithm 1 */
+    Eigen::MatrixXd Omega = W0;
+    Omega = (1.0 / sqrt(ens_size - 1.0)) *
+            (Omega.colwise() - Omega.rowwise().mean());
+    Omega.diagonal().array() += 1.0;
+
+    /* Solving for the average sensitivity matrix.
+       Line 6 of Algorithm 1, also Section 2.5 
+    */
+    Omega.transposeInPlace();
+    Eigen::MatrixXd S = Omega.fullPivLu().solve(Y.transpose()).transpose();
 
-    /* INNOVATION H = S*W + D - Y   from Eq. (41) (Line 8)*/
-    Eigen::MatrixXd H = D + S * W0; // H=D=dobs + E - Y
+    /* Similar to the innovation term.
+       Differs in that `D` here is defined as dobs + E - Y instead of just dobs + E as in the paper. 
+       Line 7 of Algorithm 1, also Section 2.6
+    */
+    Eigen::MatrixXd H = D + S * W0;
 
     /* Store previous W for convergence test */
     Eigen::MatrixXd W = W0;
 
     /*
      * COMPUTE NEW UPDATED W                                                                        (Line 9)
-     *    W = W + ies_steplength * ( W - S'*(S*S'+R)^{-1} H )          (a)
+     *    W = W - ies_steplength * ( W - S'*(S*S'+R)^{-1} H )          (a)
      * which in the case when R=I can be rewritten as
-     *    W = W + ies_steplength * ( W - (S'*S + I)^{-1} * S' * H )    (b)
+     *    W = W - ies_steplength * ( W - (S'*S + I)^{-1} * S' * H )    (b)
```

Comment:
This seems like a rather significant change, has this comment just always been wrong?

Reply:
Plus minus, potato potato!

I think the comment has been wrong, see line 8 in the figure taken from the IES-paper.

![Screenshot 2022-04-05 at 14 56 49](https://user-images.githubusercontent.com/45088507/161758946-56e2c96e-ad15-4190-8ed6-d5b38a8e1e96.png)

Reply:
Are you sure it's not a typo in the figure?

Reply:
Do you think there's a typo in the figure?